### PR TITLE
Refactor UI subject-scoped ordering to be backed by the API

### PR DESCRIPTION
**Pull request by wtritch**
Our existing orderings.ts was providing a lot of hard-coded subject/claim/etc-scoped orderings that need to be provided by the back-end API to support configurable subjects.

This PR introduces an OrderingService that provides observable orderings and refactors consumers to subscribe and sort.

### CODE REVIEW

File: webapp/src/main/webapp/src/app/aggregate-report/query-forms/claim-report-form.component.ts [hunk 1]
```diff
@@ -168,15 +172,25 @@ export class ClaimReportFormComponent extends MultiOrganizationQueryFormComponen
   }
 
   private initializeClaimsForAssessmentType(): void {
-    this.filteredOptions.subjects.forEach(subject => {
+    const orderingObservables: Observable<boolean>[] = this.filteredOptions.subjects.map(subject => {
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/aggregate-report/query-forms/claim-report-form.component.ts]: This is a little hack-tackular in that I'm applying the ordering as the observable is resolved then returning a boolean to signal when the resolution occurs.
I need call `this.initializeSelectionBySubject()` only after all of the ordering and filtering has been completed.

File: webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts [hunk 2]
```diff
@@ -30,10 +34,9 @@ const SchoolYearOrdering: Ordering<AggregateReportItem> = ordering(byNumber)
 const AssessmentLabelOrdering: Ordering<AggregateReportItem> = ordering(byString)
   .on(item => item.assessmentLabel);
 
-// TODO:ConfigurableSubjects this needs to be provided by the backend
 const OrganizationalClaimOrderingProvider: (subjectCode: string, preview: boolean) => Ordering<AggregateReportItem> = (subjectCode, preview) => {
-  const currentOrdering: Ordering<string> = !preview && SubjectClaimOrderings.has(subjectCode)
-    ? SubjectClaimOrderings.get(subjectCode)
+  const currentOrdering: Ordering<string> = !preview
+    ? getOrganizationalClaimOrdering(subjectCode)
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts]: I refactored `getOrganizationalClaimOrdering(subjectCode)` to provide the fallback to a "byString" alpha ordering if the subject isn't Math or ELA.

File: webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts [hunk 3]
```diff
@@ -384,6 +383,7 @@ export class AggregateReportTableComponent implements OnInit {
         );
         break;
       case AggregateReportType.Target:
+        this._orderingByColumnField[ 'claimCode' ] = OrganizationalClaimOrderingProvider(rows[ 0 ].subjectCode, this.preview);
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts]: organizational claim ordering is still constant (requiring no back-end calls) since we only explicitly order Math and ELA organizational claims (all other subjects have their organizational claims alpha-sorted)

File: webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts [hunk 4]
```diff
@@ -333,11 +337,6 @@ export class AggregateReportTableComponent implements OnInit {
     this._orderingByColumnField[ 'organization.name' ] = organizationOrdering(item => item.organization, rows);
     this._orderingByColumnField[ 'assessmentGradeCode' ] = assessmentGradeOrdering;
     this._orderingByColumnField[ 'schoolYear' ] = SchoolYearOrdering;
-    this._orderingByColumnField[ 'claimCode' ] = reportType === AggregateReportType.Target
-      ? OrganizationalClaimOrderingProvider(rows[ 0 ].subjectCode, this.preview)
-      : ordering(join(...rows.map(row => {
-        return createScorableClaimOrdering(row.subjectCode).on<any>(row => row.claimCode).compare;
-      })));
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts]: Small nit, this was apparently registering a joined ordering containing a comparator for each result row... this would have been pretty painful with 10k+ rows...

File: webapp/src/main/webapp/src/app/aggregate-report/query-forms/claim-report-form.component.ts (hunk 1 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/aggregate-report/query-forms/claim-report-form.component.ts]: lol what? and i thought the back button observable logic was hard to follow x J

File: webapp/src/main/webapp/src/app/shared/ordering/ordering.service.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { Injectable } from "@angular/core";
+import { SubjectService } from "../../subject/subject.service";
+import { Ordering, ordering } from "@kourge/ordering";
+import { Observable } from "rxjs/Observable";
+import { ranking } from "@kourge/ordering/comparator";
+import { map } from "rxjs/operators";
+
+@Injectable()
+export class OrderingService {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/shared/ordering/ordering.service.ts]: 👍 i like it

File: webapp/src/main/webapp/src/app/shared/ordering/orderings.ts [hunk 6]
```diff
@@ -58,7 +47,6 @@ export const byNumericString = (a: string, b: string) => {
   return a.localeCompare(b);
 };
 
-
 export function createRankingOrStringOrdering<T>(rankings: any[]): Ordering<T> {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/shared/ordering/orderings.ts]: this method i made still needed?

File: webapp/src/main/webapp/src/app/aggregate-report/query-forms/claim-report-form.component.ts (hunk 1 above)
Comment (wtritch) [reply webapp/src/main/webapp/src/app/aggregate-report/query-forms/claim-report-form.component.ts]: I know, it's a mess.  Suggestions welcome =)

File: webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts [hunk 7]
```diff
@@ -398,8 +398,35 @@ export class AggregateReportTableComponent implements OnInit {
       ...dataColumns
     ];
 
-    this.calculateTreeColumns();
-    this.sortRows();
+    if (reportType === AggregateReportType.Claim) {
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts]: If this is a claim report, we need to request scorable claim ordering based upon the subjects we're displaying.

File: webapp/src/main/webapp/src/app/assessments/model/aggregate-target-score-row.model.ts [hunk 8]
```diff
@@ -5,7 +5,6 @@ export class AggregateTargetScoreRow {
   targetId: number;
   target: string;
   claim: string;
-  claimOrder: number;
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/assessments/model/aggregate-target-score-row.model.ts]: Unused property

File: webapp/src/main/webapp/src/app/shared/ordering/orderings.ts (hunk 6 above)
Comment (wtritch) [reply webapp/src/main/webapp/src/app/shared/ordering/orderings.ts]: It's being used, but I should probably refactor it to use the OrderingService instead.

Comment (wtritch) [reply webapp/src/main/webapp/src/app/shared/ordering/orderings.ts]: refactored and removed
